<launch>
	
	<!-- launch tabletop detection service -->
	<node name="table_object_detection_node" pkg="bwi_perception" type="table_object_detection_node" output="screen">
	</node>
	
	<!-- launch tabletop detection service -->
	<node name="segbot_table_approach_as" pkg="segbot_arm_manipulation" type="segbot_table_approach_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="segbot_tabletop_grasp_as" pkg="segbot_arm_manipulation" type="segbot_tabletop_grasp_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="arm_press_as" pkg="segbot_arm_manipulation" type="arm_press_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="arm_push_as" pkg="segbot_arm_manipulation" type="arm_push_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="arm_lift_verify_as" pkg="segbot_arm_manipulation" type="arm_lift_verify_as" output="screen">
	</node>
	
	<!-- launch change detection service -->
	<node name="arm_shake_as" pkg="segbot_arm_manipulation" type="arm_shake_as" output="screen">
	</node>
	
	<!-- launch agile grasp -->
	
	<node name="find_grasps" pkg="agile_grasp" type="find_grasps" output="screen">
  	<param name="cloud_topic" value="/agile_grasp_demo/cloud" />
    <param name="cloud_frame" value="/xtion_camera_rgb_optical_frame" />
    <param name="cloud_type" value="0" />
    <param name="svm_file_name" value="$(find agile_grasp)/svm_032015_20_20_same" />
    <param name="num_samples" value="4000" />
    <param name="num_threads" value="4" />
    <param name="num_clouds" value="1" />
		<rosparam param="workspace"> [-1, 1, -1, 1, -1, 1.5] </rosparam>
    <rosparam param="camera_pose"> [0, 0.445417, 0.895323, 0.215, 
              1, 0, 0, -0.015, 
              0, 0.895323, -0.445417, 0.23, 
              0, 0, 0, 1] </rosparam>
    <param name="min_inliers" value="10" />
    <param name="plotting" value="2" />
    <param name="marker_lifetime" value="500" />
    <param name="hand_outer_diameter" value="0.10" />
    <param name="hand_depth" value="0.06" />
	</node>
	
</launch>
